pkg/operator/ceph/cluster/osd/update.go

```diff
@@ -79,6 +79,19 @@ func (c *updateConfig) updateExistingOSDs(errs *provisionErrors) {
 	if c.doneUpdating() {
 		return // no more OSDs to update
 	}
+	if !c.cluster.spec.SkipUpgradeChecks && c.cluster.spec.UpgradeOSDRequiresHealthyPGs {
+		pgHealthMsg, pgClean, err := cephclient.IsClusterClean(c.cluster.context, c.cluster.clusterInfo, c.cluster.spec.DisruptionManagement.PGHealthyRegex)
+		if err != nil {
+			logger.Errorf("failed to check PGs status to update osd, will try updating it again later. %v", err)
```

Comment:
If there is an error, Rook should return an error so that it is captured and reported to the controller. Either this function should return an error, or this logic should be moved to the routine that calls `updateExistingOSDs()`

Reply:
Agreed, let's change this to a warning

Reply:
Had a chat with Travis about this. This function doesn't return error on purpose, so that can't be done. The given means of retuning errors is through the `provisionErrors` parameter; however, that is for  And it does seem like this is the appropriate place in the code to do this check. However, I think logging errors should only be done when necessary. Our general strategy has been that if there is an error we can try to continue from, we report it as a warning, so please change this to `logger.Warningf`. For reference, `OSDUpdateShouldCheckOkToStop()` logs warnings when there are ceph cli errors as well, for this reason.